src/Controllers/CopyPatrol.php

```diff
@@ -426,30 +426,29 @@ public function getUserContribs( $user ) {
 	}
 
 	/**
-	 * Get links to compare with
-	 *
+	 * Get URLs and scores for the copyvio sources
 	 * @param $text string Blob from db
-	 * @return array matched urls
+	 * @return array Associative array with URLs and scores
 	 */
-	public function getCopyvioUrls( $text ) {
-		preg_match_all( '#\bhttps?://[^\s()<>]+(?:\([\w\d]+\)|([^[:punct:]\s]|/))#', $text, $match );
-		$uniqueCopyvioUrls = [];
-		foreach ( $match[0] as $foundUrl ) {
-			if ( !in_array( $foundUrl, $uniqueCopyvioUrls ) ) {
-				// Determine if $value is a substring of an existing url, and if so, discard it
-				// This is because of the way Plagiabot currently stores reports in its database
-				// At some point, fix this in Plagiabot code instead of this hack here
-				$isSubstring = false;
-				foreach ( $uniqueCopyvioUrls as $u ) {
-					if ( strpos( $u, $foundUrl ) !== false ) {
-						$isSubstring = true;
-					}
-				}
-				if ( $isSubstring === false ) {
-					$uniqueCopyvioUrls[] = $foundUrl;
-				}
-			}
+	public function getSources( $text ) {
+		// matches '[new line] ... (digits followed by %) (digits) ... (the URL)[word break]'
+		preg_match_all( '#\n\*.*?(\d+%)\s+(\d+).*?\b(https?://[^\s()<>]+)\b#', $text, $matches );
```

Comment:
Here I simplified the old regex capturing the URL. It's unclear to what `[:punct:]` does or why we need it, but from my testing the above regex appears to work. Also I'm not sure what the comments `Determine if $value is a substring of an existing url, and if so, discard it` mean. I omitted this "hack" and all seems to work fine, but I'm probably wrong... are there any examples of where we need this?